lib/printer.js

```diff
@@ -954,6 +1006,9 @@ function genericPrintNoParens(path, options, print) {
     case "Specifier":
     case "NamedSpecifier":
     case "Comment": // Supertype of Block and Line.
+    case "MemberTypeAnnotation": // Flow
+    case "TupleTypeAnnotation": // Flow
+    case "Type": // Flow
```

Comment:
@cpojer  thanks for your reply, created an issue here to track it (https://github.com/benjamn/recast/issues/422)